### ISSUE TITLE
Inline never works with Inline:MakeMaker for Pdlpp and others

### ISSUE THREAD

**Issue by leonid-butenko** [ingydotnet/inline-pm#55]
Hi there, 

I think I have discovered a bug in Inline perl module. In Inline.pm's install routine (version 0.55), there is a check for language name which is apparently wrong for most of the languages:

http://cpansearch.perl.org/src/SISYPHUS/Inline-0.55/Inline.pm:

    croak M64_install_not_c($o->{API}{language_id})
        unless uc($o->{API}{language_id}) =~ /^(C|CPP|Java|Python|Ruby|Lisp|Pdlpp)$/ ;

Upper case will never match Python, Pdlpp etc. Pls advise..

regards,
Leo

**Comment by mohawk2** [comment]
You are correct. The "install" functionality of `Inline` has never worked brilliantly. Check out `Inline::Module` instead.

**Comment by leonid-butenko** [comment]
Hey, thanks for the advice ... I tried to use Inline::Module instead but it doesn't look one to one replacement. Can you suggest about the following problem?

If i generate a tarball from my package using make dist the inline stub generated is getting substituted with the DynaLoader version. Although if I'm running make install - it keeps using the standard one like this:

use strict; use warnings;
package pdlinc::Inline;
use Inline::Module stub => 'v2';

To my understanding this should work same as make dist, shouldn't it ?

**Comment by mohawk2** [comment]
The above, if I understand you right, is all build-time stuff. The purpose of I::M is that after it is installed, only the DynaLoader will be in the installed version, and it will all "just work". Have you tried that? You'd probably be best off using a "perlbrew"-ed perl to try it, rather than your system perl.
